bin/improver-gradient

```diff
@@ -0,0 +1,71 @@
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to calculate gradient of input field in x and y direction."""
+
+import argparse
+import os
+import iris
+
+from improver.utilities.spatial import DifferenceBetweenAdjacentGridSquares
+from improver.generate_ancillaries.generate_ancillary import (
+    find_standard_ancil)
+
+
+def main():
+    """Load in arguments and get going."""
+    parser = argparse.ArgumentParser(
+        description=('Read the input field, and calculate the gradient'
+                     ' in x and y directions.'))
+    parser.add_argument('--force', dest='force', default=False,
+                        action='store_true',
+                        help=('If True, ancillaries will be generated '
+                              'even if doing so will overwrite existing '
+                              'files.'))
+    parser.add_argument('input_filepath_standard',
+                        metavar='INPUT_FILE_STANDARD',
+                        help='A path to an input NetCDF file to be processed')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILE',
+                        help='The output path for the processed NetCDF')
+    args = parser.parse_args()
+    # Check if improver ancillary already exists.
+    if not os.path.exists(args.output_filepath) or args.force:
+        input_field = find_standard_ancil(args.input_filepath_standard)
```

Comment:
I'm not sure there's a reason to be using `find_standard_ancil` rather than just using `iris.load_cube`. I think that `find_standard_ancil` is a basically a precursor to the Load plugin desired in #31. As this `gradient` CLI is designed to be of general usage, it's not great to be importing something specific from `improver.generate_ancillaries.generate_ancillary`.